emergence_lib/src/filtered_array_iter.rs

```diff
@@ -0,0 +1,67 @@
+//! Allocation-less struct to non-lazily filtered array iterators
+
+use std::{array, iter};
+
+/// Allocation-less struct to non-lazily filtered array iterators
```

Comment:
```suggestion
/// Allocation-less struct to non-lazily filter array iterators
```

Reply:
I think this was the intent? The comment wasn't super clear to me.

Reply:
It was... I struggled to find a reasonable way to describe and motivate the struct in the comment to be honest.